toolchain/check/eval.cpp

```diff
@@ -1311,6 +1324,41 @@ static auto MakeConstantForBuiltinCall(Context& context, SemIRLoc loc,
       return SemIR::ConstantId::NotConstant;
     }
 
+    case SemIR::BuiltinFunctionKind::TypeAnd: {
+      CARBON_CHECK(arg_ids.size() == 2);
+      auto facet_type_ids = std::to_array<SemIR::FacetTypeId>({
+          SemIR::FacetTypeId::None,
+          SemIR::FacetTypeId::None,
+      });
```

Comment:
I'm trying to understand why you're using arrays here. Also, why do you use `llvm::any_of` on line 1346 when there are only two values, versus `!facet_type_ids[0].has_value() || !facet_type_ids[1].has_value()`?

If this is mainly for code-sharing of 1334-1342, had you considered instead declaring `lhs` and `rhs`, and either something like:

```
for (auto [facet_type_id, arg_id] : llvm::zip(std::to_array({&lhs_id, &rhs_id}), arg_ids)) {
```

Or a lambda like:

```
auto get_facet_type_id = [](SemIR::InstId arg_id) ...
auto lhs_id = get_facet_type_id(arg_ids[0]);
auto rhs_id = get_facet_type_id(arg_ids[1]);
```

(note, I commented on a prior PR about single-use lambdas, but here it'd support code reuse)

Reply:
I did not consider that zip, but it's nice, thanks.